cmd/node/factory/structs.go

```diff
@@ -981,93 +991,176 @@ func createShardDataStoreFromConfig(
 		}
 	}()
 
-	txUnit, err = storageUnit.NewStorageUnitFromConf(
-		getCacherFromConfig(config.TxStorage.Cache),
-		getDBFromConfig(config.TxStorage.DB, uniqueID),
-		getBloomFromConfig(config.TxStorage.Bloom))
+	fullArchiveMode := config.StoragePruning.FullArchive
```

Comment:
this function might be put inside a factory - and deleted from structs.go

Reply:
done. created a factory for shard and meta storage services